internal/store/serverservice.go

```diff
@@ -0,0 +1,127 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/coreos/go-oidc"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/metal-toolbox/hollow-bomservice/internal/app"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+
+	sservice "go.hollow.sh/serverservice/pkg/api/v1"
+)
+
+// Serverservice implements the Repository interface to have server bom objects stored as server attributes.
+type Serverservice struct {
+	config *app.ServerserviceOptions
+	client *sservice.Client
+	logger *logrus.Logger
+}
+
+var (
+	// ErrServerserviceConfig is returned when theres an error in loading serverservice configuration.
+	ErrServerserviceConfig = errors.New("Serverservice configuration error")
+
+	// ErrServerserviceQuery is returned when a serverservice query error was received.
+	ErrServerserviceQuery = errors.New("Serverservice query error")
+
+	// ErrServserviceAttribute is returned when a serverservice attribute does not contain the expected fields.
+	ErrServerserviceAttribute = errors.New("error in serverservice attribute")
+
+	// connectionTimeout is the maximum amount of time spent on each http connection to serverservice.
+	connectionTimeout = 30 * time.Second
+)
+
+func newServerserviceStore(ctx context.Context, config *app.ServerserviceOptions, logger *logrus.Logger) (Repository, error) {
+	s := &Serverservice{logger: logger, config: config}
+
+	var client *sservice.Client
+	var err error
+
+	if !config.DisableOAuth {
+		client, err = newClientWithOAuth(ctx, config, logger)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client, err = sservice.NewClientWithToken("fake", config.Endpoint, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	s.client = client
+
+	return s, nil
+}
+
+// returns a serverservice retryable http client with Otel and Oauth wrapped in
+func newClientWithOAuth(ctx context.Context, cfg *app.ServerserviceOptions, logger *logrus.Logger) (*sservice.Client, error) {
+	fmt.Println("newClientWithOAuth")
```

Comment:
You should get rid of this `Println`. If you want to track method entry/exit, log it at `Trace`.